scipy/sparse/csgraph/tests/test_tools.py

```diff
@@ -0,0 +1,148 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_equal
+
+from scipy.sparse.csgraph import (
+    csgraph_from_adjacency_list,
+    csgraph_to_adjacency_list,
+    csgraph_from_dense,
+)
+
+
+def test_csgraph_weighted_adjacency_list():
```

Comment:
Explicit zeros (i.e. instances of `csr_matrix` whose `data` contains one or more `0`s) tend to lead to a lot of gotchas, so it could be useful to have tests for those (in both directions).